### PR TITLE
api: Add const BString::new

### LINKED ISSUE #73: api: const fn BString::new
I'm using `BString` to implement an output capture struct and leverage the `BString` debug impl.

I'm currently adding `const` to as many zero argument `new` constructors in my code as I can. Since `BString` is a thin wrapper around `Vec`, I expected to have an analogous `pub const fn new()` method available.

There is no workaround for this since `Into::into` is not const.

### PR DESCRIPTION
**Pull request by lopopolo**
Fixes #73.

### CODE REVIEW

File: src/bstring.rs [hunk 1]
```diff
@@ -44,8 +44,23 @@ pub struct BString {
 }
 
 impl BString {
+    /// Constructs a new `BString` from the given `Vec`.
```
Comment (lopopolo) [inline src/bstring.rs:47]:
```suggestion
    /// Constructs a new `BString` from the given [`Vec`].
```

File: src/bstring.rs [hunk 2]
```diff
@@ -44,8 +44,23 @@ pub struct BString {
...
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use bstr::BString;
```
Comment (BurntSushi) [inline src/bstring.rs:52]: I think the examples should include `use` statements. The one below as well.

**Review by BurntSushi — CHANGES_REQUESTED** [review]
LGTM with a small nit!

File: src/bstring.rs (hunk 2 above)
Comment (lopopolo) [reply src/bstring.rs]: fixed 😄

**Comment by lopopolo** [comment]
Thanks for the review @BurntSushi. I've force pushed with your suggested changes.

**Review by BurntSushi — APPROVED** [review]
Thanks!

**Pull request merged**